components/content-service/pkg/archive/tar.go

```diff
@@ -125,23 +124,23 @@ func ExtractTarbal(ctx context.Context, src io.Reader, dst string, opts ...TarOp
 		paths = append(paths, path)
 	}
 	sort.Sort(sort.Reverse(sort.StringSlice(paths)))
+	/*
+		// We need to remap the UID and GID between the host and the container to avoid permission issues.
+		for _, p := range paths {
+			v := m[p]
+			uid := toHostID(v.UID, cfg.UIDMaps)
+			gid := toHostID(v.GID, cfg.GIDMaps)
+
+			if v.IsSymlink {
+				continue
+			}
 
-	// We need to remap the UID and GID between the host and the container to avoid permission issues.
-	for _, p := range paths {
-		v := m[p]
-		uid := toHostID(v.UID, cfg.UIDMaps)
-		gid := toHostID(v.GID, cfg.GIDMaps)
-
-		if v.IsSymlink {
-			continue
-		}
-
-		err = remapFile(path.Join(dst, p), uid, gid)
-		if err != nil {
-			log.WithError(err).WithField("uid", uid).WithField("gid", gid).WithField("path", p).Warn("cannot chown")
+			err = remapFile(path.Join(dst, p), uid, gid)
+			if err != nil {
+				log.WithError(err).WithField("uid", uid).WithField("gid", gid).WithField("path", p).Warn("cannot chown")
+			}
```

Comment:
I'm sure there are architectural reasons for this, but as you already know, the way to get a predictable tar unarchive is to extract with root privs on the system it's targeted for. That's far simpler than trying to maintain a mapping of uid/gid between two systems, which is almost certain to fail. But again, I know you're up against things that I have no idea about.

Reply:
@rfay the issue is not located in the restoration of the backup but in the creation. In particular, [this is the fix](https://github.com/gitpod-io/gitpod/pull/3636/files#diff-8cf81f14073bcad3e67f46c7667d82a56824f72ddffb9529e71944719e44a4afR163-R165)
If this works correctly I will refactor the `remapFile` helper